crates/nu-cli/src/commands/to_md.rs

```diff
@@ -28,39 +37,108 @@ impl WholeStreamCommand for ToMarkdown {
     ) -> Result<OutputStream, ShellError> {
         to_md(args, registry).await
     }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Outputs an unformatted md string representing the contents of ls",
+                example: "ls | to md",
+                result: None,
+            },
+            Example {
+                description: "Outputs a formatted md string representing the contents of ls",
+                example: "ls | to md -p",
+                result: None,
+            },
+        ]
+    }
 }
 
 async fn to_md(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream, ShellError> {
     let registry = registry.clone();
-    let args = args.evaluate_once(&registry).await?;
-    let name_tag = args.name_tag();
-    let input: Vec<Value> = args.input.collect().await;
+    let name_tag = args.call_info.name_tag.clone();
+    let (ToMarkdownArgs { pretty }, input) = args.process(&registry).await?;
+    let input: Vec<Value> = input.collect().await;
     let headers = nu_protocol::merge_descriptors(&input);
     let mut output_string = String::new();
 
+    let mut column_length_vector: Vec<usize> = Vec::new();
+
+    if pretty {
+        if !headers.is_empty() && (headers.len() > 1 || headers[0] != "") {
+            for header in &headers {
+                let htmlescape_header_string = &htmlescape::encode_minimal(&header);
+                column_length_vector.push(htmlescape_header_string.len());
+            }
+        }
+
+        for row in &input {
+            match row.value.clone() {
+                UntaggedValue::Row(row) => {
+                    for i in 0..headers.len() {
+                        let data = row.get_data(&headers[i]);
+                        let new_column_length =
+                            format_leaf(data.borrow()).plain_string(100_000).len();
+
+                        if column_length_vector[i] < new_column_length {
+                            column_length_vector[i] = new_column_length;
+                        }
+                    }
+                }
+                _ => {}
+            }
+        }
```

Comment:
This chunk is the part that bothers me the most.  It is essentially a copy and paste of the structure that was already there originally, but it basically pre-computes the longest strings in each column.  Its pretty gross because it does the exact same thing as what will occur right after when the values start getting added to the output string.  The only thing I can think to do is to calculate the width of each column at the same time as the htmlescape encoding.  The htmlescape values can then be stored in a new data structure that can be iterated over again.  This solution still feels a bit unclean, so I decided not to jump on it until I heard from others.